### PR TITLE
Automate the release process

### PR DESCRIPTION
**Pull request by bokay**
## Background

Currently the release process requires a number of manual steps such as creating the release on GitHub. Since there are manual steps, this can be error prone.

## Requirement

Eliminate as many manual steps as possible:

- [ ] Create the release on GitHub via the Makefile target

### CODE REVIEW

**Comment by bokay** [comment]
@frahugo FYI

```shell
➜  platform git:(chore/automate-release) time git tag | tail -n1
release-201902a
git tag  0.00s user 0.01s system 41% cpu 0.030 total
tail -n1  0.00s user 0.00s system 13% cpu 0.028 total

➜  platform git:(chore/automate-release) time hub release -L 1
release-201902a
hub release -L 1  0.08s user 0.04s system 19% cpu 0.612 total
```

`hub` seems to systematically make a network call, so using `git tag` will always be faster.

**Comment by bokay** [comment]
@frahugo @nicholasjhenry This is ready for another pass.

File: bin/generate_release [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    if [ "$1" == "$2" ]; then
+        echo "$3" | tr "a-z" "b-z"
+    else
+        echo "a"
+    fi
+}
+
+# Retrieve previous release tag
+git fetch --all --tags
+PREVIOUS_RELEASE_TAG=`git tag | tail -n1`
```
Comment (frahugo) [inline bin/generate_release]: Use 'hub-release' to get the last release. Using `git tag` is not safe as we could have tags other than release tags in the repo.

File: bin/generate_release [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+# Create a new empty release
+hub release create -m "${RELEASE_TAG}" $RELEASE_TAG
+
+git fetch --all --tags
+git checkout master
+git pull -f
+
+# Edit current release message
+echo "${RELEASE_TAG}" > RELEASE_NOTES
```
Comment (frahugo) [inline bin/generate_release]: Not sure we need this in the release description as the release already has a title.

File: Makefile [hunk 3]
```diff
@@ -52,3 +52,6 @@ observer:
 
 console:
 	docker-compose exec application iex --name vm@application --cookie secret -S mix phx.server
+
+prepare_release:
```
Comment (frahugo) [inline Makefile]: Maybe `generate_release` instead? like the script name?

File: bin/generate_release (hunk 2 above)
Comment (bokay) [reply bin/generate_release]: The first line is used to modify the title

**Pull request merged**